x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/filestructurefinder/FieldStats.java

```diff
@@ -142,6 +142,15 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         return builder;
     }
 
+    public static Object toIntegerIfInteger(double d) {
+
+        if (d >= Integer.MIN_VALUE && d <= Integer.MAX_VALUE && Double.compare(d, StrictMath.rint(d)) == 0) {
```

Comment:
#fun-note

Reading up on how to do this operation, there is also a clever (maybe too clever) way of `(d % 1) == 0` instead of `Double.compare(d, StrictMath.rint(d)) == 0`, though your version is probably more clear :D

Reply:
👍